src/Silverback.Integration.Kafka/Messaging/Broker/KafkaConsumer.cs

```diff
@@ -379,6 +380,8 @@ private void InitConfluentConsumer()
 
                 Endpoint.TopicPartitions.ForEach(
```

Comment:
If I unterstand correctly, if subscribing the channel manager is set the partitions are assigned.
Wouldn't it be more elegant to also use the same mechanism here and do not distinguierst these cases?